src/app/qgisapp.cpp

```diff
@@ -12241,9 +12243,15 @@ void QgisApp::removeLayer()
     return;
   }
 
-  const auto constSelectedNodes = selectedNodes;
-  for ( QgsLayerTreeNode *node : constSelectedNodes )
+  for ( QgsLayerTreeNode *node : selectedNodes )
   {
+    if ( QgsLayerTreeGroup *group = qobject_cast< QgsLayerTreeGroup * >( node ) )
+    {
+      if ( QgsGroupLayer *groupLayer = group->groupLayer() )
+      {
+        QgsProject::instance()->removeMapLayer( groupLayer );
```

Comment:
could this be an issue regarding the order of signal here ?
`layerWillBeRemoved` and `layerRemoved` will both be sent first for the layer group, and later for each layer.

Reply:
I think that should be fine -- the group layers don't have any ownership of the other layers, so to me it makes sense that they get cleaned up first

Reply:
ok, thanks for the answer